tests/gflownet/envs/common.py

```diff
@@ -204,7 +204,15 @@ def test__get_parents__returns_same_state_and_eos_if_done(env):
     env.set_state(env.state, done=True)
     parents, actions = env.get_parents()
     if torch.is_tensor(env.state):
-        assert all([torch.equal(p, env.state) for p in parents])
+        assert all(
+            [torch.equal(torch.isnan(p), torch.isnan(env.state)) for p in parents]
+        )
+        assert all(
+            [
+                torch.equal(p[~torch.isnan(p)], env.state[~torch.isnan(env.state)])
+                for p in parents
+            ]
+        )
```

Comment:
Why was this needed?

Reply:
Ah it's because the states of the tree are initialised to `torch.nan`?

Reply:
Yes, and (get ready to be amazed):
```
>>> torch.nan == torch.nan
False
```

Reply:
LOL

Reply:
I guess it makes sense (ish) if I think about it... At least it's consistent with numpy:
```
>>> numpy.inf * 0 == numpy.inf * 0
False
```

Reply:
Well riddle me this:
```
>>> torch.inf == torch.inf
True
```
and
```
>>> torch.nan in [torch.nan]
True
```

Reply:
hahaha I can see the logic of either way...

Reply:
I have incorporated the check of NaNs in `env.equal(x, y)` so it's easier to check if two states are the same also with nans. See [here](https://github.com/alexhernandezgarcia/gflownet/commit/c347096a001a36858d99b13f6c3223b66ddff2b0)